server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -41,6 +42,19 @@
     private ReportType type = ReportType.CPU;
     private boolean ignoreIdleThreads = true;
 
+    private static final List<String[]> knownElasticIdleFrames = Arrays.asList(
+        new String[] {"java.util.concurrent.ThreadPoolExecutor", "getTask"},
+        new String[] {"sun.nio.ch.SelectorImpl", "select"},
+        new String[] {"org.elasticsearch.threadpool.ThreadPool$CachedTimeThread", "run"},
+        new String[] {"org.elasticsearch.indices.ttl.IndicesTTLService$Notifier", "await"},
+        new String[] {"java.util.concurrent.LinkedTransferQueue", "poll"}
+    );
+
+    // NOTE: these are JVM dependent and JVM version dependent
+    private static final List<String> knownJvmInternalThreads = Arrays.asList(
+        "Signal Dispatcher", "Finalizer", "Reference Handler", "Notification Thread", "Common-Cleaner"
+    );
```

Comment:
This list seems ok. Common-Cleaner is a "JDK-libraries" thread rather than a JVM thread per se, but filtering it seems reasonable. We could additionally filter "process reaper", or "JMX server connection timeout XX", but these threads shouldn't be doing much either way.

Reply:
Great! I added "process reaper" in the list, while since the JMX thread name is numbered, I used the idle stack frame list to catch that one.